spec/std/enumerable_spec.cr

```diff
@@ -91,19 +91,23 @@ describe "Enumerable" do
 
   describe "compact" do
     it "can be applied to an Array" do
-      [nil, 1, nil, 2, nil, 3, nil].compact.should eq([1, 2, 3])
+      coll = [nil, 1, nil, 2, nil, 3, nil].compact
+      coll.to_a.should eq([1, 2, 3])
```

Comment:
Here we're testing the concrete implementation of `Enumerable#compact`. IMO it makes sense to expect the method to return `Array`.

Reply:
(the same applies to the tests below)